app/frontend/stores/user-store.ts

```diff
@@ -0,0 +1,49 @@
+import { values } from "mobx"
+import { Instance, flow, types } from "mobx-state-tree"
+import * as R from "ramda"
+import { withEnvironment } from "../lib/with-environment"
+import { withRootStore } from "../lib/with-root-store"
+import { IUser, UserModel } from "../models/user"
+// import { IUser, UserModel } from "../models/user"
+
+export const UserStoreModel = types
+  .model("UserStoreModel")
+  .props({
+    usersMap: types.map(UserModel),
+    currentUser: types.maybeNull(types.safeReference(UserModel)),
+  })
+  .extend(withEnvironment())
+  .extend(withRootStore())
+  .views((self) => ({
+    get users(): IUser[] {
+      //@ts-ignore
+      return values(self.usersMap) as IUser[]
+    },
+  }))
+  .views((self) => ({
+    getUser(id) {
+      return self.users.find((u) => u.id == id)
+    },
+    isCurrentUser(user) {
+      return user.id == self.currentUser.id
+    },
```

Comment:
are these being used? Let's avoid adding extra code unless it's needed as this increases the chance of duplications / stale code. Same with the actions below.